braze/src/main/java/com/rudderstack/android/integration/braze/BrazeIntegrationFactory.java

```diff
@@ -234,48 +298,92 @@ private void processTrackEvent(RudderMessage element) {
+                                (String) campaignProps.get("name"),
+                                (String) campaignProps.get("ad_group"),
+                                (String) campaignProps.get("ad_creative")));
+                    }
+                } else {
+                    logCustomEvent(event, eventProperties);
                 }
-                return;
+            } else if (event.equals("Order Completed")) {
+                List<BrazePurchase> purchaseList = getPurchaseList(eventProperties);
+                if (purchaseList != null) {
+                    for (BrazePurchase purchase: purchaseList) {
+                        this.braze.logPurchase(purchase.getProductId(), purchase.getCurrency(),
+                                BigDecimal.valueOf(purchase.getPrice()), new BrazeProperties(purchase.properties));
+                        RudderLogger.logDebug(String.format("Braze logPurchase for purchase %s for %s %s %s", purchase.getProductId(),
+                                purchase.getPrice(), purchase.getCurrency(), purchase.properties.toString()));
+                    }
+                }
+            } else {
+                logCustomEvent(event, eventProperties);
             }
         } catch (Exception exception) {
             RudderLogger.logError(exception);
         }
+    }
 
-        if (eventProperties == null || eventProperties.size() == 0) {
-            RudderLogger.logDebug("Braze event has no properties");
-            this.braze.logCustomEvent(element.getEventName());
-            return;
+    private void logCustomEvent(String eventName, Map<String, Object> eventProperties) {
+        if (eventProperties != null) {
+            RudderLogger.logDebug(String.format("Braze logCustomEvent for event %s with properties %% %s", eventName, eventProperties.toString()));
+            this.braze.logCustomEvent(eventName, new BrazeProperties(eventProperties));
+        } else {
+            RudderLogger.logDebug(String.format("Braze logCustomEvent for event %s", eventName));
+            this.braze.logCustomEvent(eventName);
         }
+    }
 
-        JSONObject propertiesJson = new JSONObject(eventProperties);
-        if (eventProperties.containsKey(REVENUE_KEY) && eventProperties.get(REVENUE_KEY) != null) {
-            double revenue = Double.parseDouble(String.valueOf(eventProperties.get(REVENUE_KEY)));
-            String currency = String.valueOf(eventProperties.get(CURRENCY_KEY));
-            String currencyCode = TextUtils.isEmpty(currency) ? DEFAULT_CURRENCY_CODE
-                    : currency;
-            propertiesJson.remove(REVENUE_KEY);
-            propertiesJson.remove(CURRENCY_KEY);
+    @Nullable
+    private List<BrazePurchase> getPurchaseList(Map<String, Object> properties) {
+        if (properties == null) {
+            return null;
+        }
+        List<Map<String, Object>> productList = (List<Map<String, Object>>)properties.get(PRODUCTS_KEY);
```

Comment:
As per our docs, we support products in the list only. We shouldn't keep this.

Reply:
@desusai7 What's your opinion on this?
Going forward. should we plan to handle the products as an `array` for the `Order Completed` event?
Or, should we provide the support (i.e., handle dictionary/map value) that we are doing for other destinations till now?

Reply:
For Order Completed, lets support the products array itself.  And even in our spec its always an array of products for order completed event.

And talking about the other destinations that might be because of compatibility with other tools, but since we are doing this from scratch by ourself sticking to our spec should be good enough